Subject: Check for existence of osb in kill_sb
Reference: bnc#552602
From: Goldwyn Rodrigues <rgoldwyn@novell.com>

In case of an error while mounting, for example - slots full, the volume
dismounts and calls kill_sb. However, osb is not initialized as yet,
and results in a NULL pointer dereference. Check if osb is initialized
before operating on it.

Signed-off-by: Goldwyn Rodrigues <rgoldwyn@novell.com>
Signed-off-by: Coly Li <coly.li@suse.de>
---
 ocfs2/super.c |   13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

Index: fs/ocfs2/super.c
===================================================================
--- a/fs/ocfs2/super.c	2009-11-05 00:59:58.000000000 +0100
+++ b/fs/ocfs2/super.c	2009-11-05 17:49:51.000000000 +0100
@@ -1019,12 +1019,13 @@ static void ocfs2_kill_sb(struct super_b
 	struct ocfs2_super *osb = OCFS2_SB(sb);
 
 	/* Prevent further queueing of inode drop events */
-	spin_lock(&dentry_list_lock);
-	ocfs2_set_osb_flag(osb, OCFS2_OSB_DROP_DENTRY_LOCK_IMMED);
-	spin_unlock(&dentry_list_lock);
-	/* Wait for work to finish and/or remove it */
-	cancel_work_sync(&osb->dentry_lock_work);
-
+	if (osb) {
+		spin_lock(&dentry_list_lock);
+		ocfs2_set_osb_flag(osb, OCFS2_OSB_DROP_DENTRY_LOCK_IMMED);
+		spin_unlock(&dentry_list_lock);
+		/* Wait for work to finish and/or remove it */
+		cancel_work_sync(&osb->dentry_lock_work);
+	}
 	kill_block_super(sb);
 }
 
